app/models/search.rb

```diff
@@ -62,6 +70,34 @@ def start! search_id
       search.pseudo_graph_pattern.update(started_at: Time.now.utc)
       search
     end
+
+    private
+
+    # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+    def hash_user_search search
+      pgp = search.pseudo_graph_pattern
+      {
+        search_id: search.search_id,
+        query: search.query,
+        read_timeout: pgp.read_timeout,
+        sparql_limit: pgp.sparql_limit,
+        answer_limit: pgp.answer_limit,
+        target: pgp.target,
+        private: pgp.private,
+        state: pgp.state,
+        created_at: to_strftime(pgp.created_at&.in_time_zone),
+        started_at: to_strftime(pgp.started_at&.in_time_zone),
+        finished_at: to_strftime(pgp.finished_at&.in_time_zone),
+        referred_at: to_strftime(search.referred_at&.in_time_zone),
+        elapsed_time: pgp.elapsed_time&.to_f&.ceil(1),
+        number_with_precision: pgp.answers.size
+      }
```

Comment:
今更気が付きました。
pgpの値を返す部分はpgpに移動するのが良さそうです。
```suggestion
      {
        search_id: search.search_id,
        query: search.query
      }.merge pgp.data_for_search_detail
```

Reply:
いいですね。
これで対応しました。